File: WebKBR.Domain/Models/Order.cs
using System;
using System.Collections.Generic;

namespace WebKBR.Domain.Models;

public partial class Order
{
    public int ProductId { get; set; }

    public int ClientId { get; set; }

    public int OrderId { get; set; }

    public int ModelId { get; set; }

    public int ColorId { get; set; }

    public int MdfId { get; set; }

    public short Width { get; set; }

    public short Height { get; set; }

    public decimal OrderValue { get; set; }

    public virtual Client Client { get; set; } = null!;

    public virtual Color Color { get; set; } = null!;

    public virtual Mdf Mdf { get; set; } = null!;

    public virtual Model Model { get; set; } = null!;
}

File: flutter_2kbr/lib/data/models/order.dart
class Order {
  String model;
  String color;
  String mdf;
  int height;
  int width;
  String type;
  int modelId;
  int colorId;
  int mdfId;

  Order({
    required this.model,
    required this.color,
    required this.mdf,
    required this.height,
    required this.width,
    required this.type,
    required this.modelId,
    required this.colorId,
    required this.mdfId,
  });

  Order copyWith({
    String? model,
    String? color,
    String? mdf,
    int? height,
    int? width,
    String? type,
    int? modelId,
    int? colorId,
    int? mdfId,
  }) {
    return Order(
      model: model ?? this.model,
      color: color ?? this.color,
      mdf: mdf ?? this.mdf,
      height: height ?? this.height,
      width: width ?? this.width,
      type: type ?? this.type,
      modelId: modelId ?? this.modelId,
      colorId: colorId ?? this.colorId,
      mdfId: mdfId ?? this.mdfId,
    );
  }

  Order.fromJson(Map<String, dynamic> json)
      : model = json['model'],
        color = json['color'],
        mdf = json['mdf'],
        height = json['height'],
        width = json['width'],
        type = json['type'],
        modelId = json['modelId'],
        colorId = json['colorId'],
        mdfId = json['mdfId'];

  Map<String, dynamic> toJson() => {
        'model': model,
        'color': color,
        'mdf': mdf,
        'height': height,
        'width': width,
        'type': type,
        'modelId': modelId,
        'colorId': colorId,
        'mdfId': mdfId,
      };
}
